### PR TITLE
Site Editor: Add the Post Author component to the Page sidebar

### PR DESCRIPTION
**Pull request by youknowriad**
Related #52632 

## What?

This PR adds the "post author" control from the post editor to the site editor sidebar. The goal is to ultimately bridge the gap between these two editors and just merge them.

## Testing Instructions

1- Open a page in the site editor.
2- Notice that you can change the author from the sidebar.

### CODE REVIEW

File: packages/edit-site/src/components/sidebar-edit-mode/page-panels/page-summary.js [hunk 1]
```diff
@@ -18,7 +22,7 @@ export default function PageSummary( {
 	postType,
 } ) {
 	return (
-		<VStack>
+		<VStack spacing={ 0 }>
```
Comment (youknowriad) [inline packages/edit-site/src/components/sidebar-edit-mode/page-panels/page-summary.js:25]: This small change is independent but it just removes the spacing between the different controls to match the post editor.

File: packages/editor/src/components/post-panel-row/index.js [hunk 2]
```diff
@@ -15,18 +15,10 @@ const PostPanelRow = forwardRef( ( { className, label, children }, ref ) => {
...
-					<div className="editor-post-panel__row-control">
-						{ children }
-					</div>
-				</>
-			) : (
-				children
+			{ label && (
+				<div className="editor-post-panel__row-label">{ label }</div>
 			) }
+			<div className="editor-post-panel__row-control">{ children }</div>
```
Comment (youknowriad) [inline packages/editor/src/components/post-panel-row/index.js:21]: The change here is a simplification that also makes the "pending" toggle look better spaced in the post editor.

**Comment by jameskoster** [comment]
Working for me 👍 

Screenshot 2023-11-21 at 11 46 10


It might be nice to use the same format as the other summary details, IE:

Screenshot 2023-11-21 at 11 47 14


Appreciate if that's something to handle separately though.

**Review by jameskoster — APPROVED** [review]

**Comment by youknowriad** [comment]
I agree, I don't like the design that looks so different. But I'm going to leave that for a separate issue/person probably. I'd like to move forward with more unification of existing stuff.

**Pull request merged**